### PR TITLE
Resource: nginx_install

### PR DESCRIPTION
**Pull request by fernandoaleman**
# Resource: nginx_install #486 

- Update kitchen to use chef solo
- Support only OS with systemd
- Drop support for Amazon Linux, CentOS 6
- Add support for Amazon Linux 2
- Remove community_cookbook_releaser gem
- Move spec/libraries to spec/unit/libraries
- Bump ohai dependency to ~> 5.2
- Create nginx_install custom resource with source properties distro, repo, epel and passenger
- Create nginx_config custom resource
- Update nginx_site resource
- Update default-site.erb template
- Default recipe calls nginx_install resource

## Issues Resolved

Can now install nginx via resource vs recipes.

## Check List

- [ ] All tests pass. See TESTING.md for details
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable

## TODO

- Decide if we are going to keep the option to install nginx from `source`. There was a discussion about this in the slack channel, so just have to confirm.

- The kitchen.yml still includes the `source` and `modules` tests and those are currently failing, pending the decision to keep or remove the option to build from source.

This is my first time submitting a PR to the `sous-chef` group so I look forward to any feedback and suggestions on improving the PR.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,164 +24,147 @@ workflows:
...
-          suite: distro-packages-ubuntu-1604
+          name: distro-debian-9
+          suite: distro-debian-9
           requires:
             *lint_and_unit
       - kitchen/dokken-single:
-          name: distro-packages-ubuntu-1604
-          suite: distro-packages-ubuntu-1604
+          name: distro-fedora-28
+          suite: distro-fedora-28
```
Comment (damacus) [inline .circleci/config.yml]: It's good but I don't think we need Fedora here :)

File: .kitchen.dokken.yml [hunk 2]
```diff
@@ -1,18 +1,22 @@
 ---
 driver:
   name: dokken
-  privileged: true  # because Docker and SystemD/Upstart
+  # because Docker and SystemD/Upstart
+  privileged: true
+  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>
```
Comment (damacus) [inline .kitchen.dokken.yml]: The old way was the correct way. Or at least the new recommended way. I'd like to roll the provisioner stuff out everywhere eventually 👍 

https://docs.chef.io/config_yml_kitchen.html

File: .kitchen.dokken.yml [hunk 3]
```diff
@@ -27,24 +31,15 @@ platforms:
...
 
-  - name: ubuntu-16.04
+  - name: fedora-28
     driver:
-      image: dokken/ubuntu-16.04
-      pid_one_command: /sbin/init
-      intermediate_instructions:
-        - RUN /usr/bin/apt-get update
+      image: dokken/fedora-28
+      pid_one_command: /usr/lib/systemd/systemd
```
Comment (damacus) [inline .kitchen.dokken.yml]: Fedora is a horrible moving target that never really gets fixes, and increases "bugs" in the cookbooks. 

We should put this in our official docs that we don't support Fedora, but it *probably* works

File: .kitchen.dokken.yml [hunk 4]
```diff
@@ -60,7 +55,7 @@ platforms:
       intermediate_instructions:
         - RUN /usr/bin/apt-get update
 
-  - name: amazonlinux
+  - name: opensuse-leap-42
```
Comment (damacus) [inline .kitchen.dokken.yml]: leap-15 is the latest. Don't ask 😆

File: resources/install.rb [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+      plugin 'nginx'
+      action :nothing
+    end
+
+    ohai_plugin 'nginx' do
+      cookbook    'nginx'
+      source_file 'plugins/ohai-nginx.rb.erb'
+      resource    :template
+      variables(
+        binary: nginx_binary
```
Comment (damacus) [inline resources/install.rb]: This feels like an overuse/premature use of variables

File: spec/support/install.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+def repo_url
+  case chefspec_platform
+  when 'amazon', 'fedora'
```
Comment (damacus) [inline spec/support/install.rb]:
```suggestion
  when 'amazon', 'fedora', 'redhat'
```

File: test/cookbooks/test/recipes/epel.rb [hunk 7]
```diff
@@ -1,4 +1,3 @@
-node.default['nginx']['repo_source'] = 'epel'
+apt_update 'update' if platform_family?('debian')
```
Comment (damacus) [inline test/cookbooks/test/recipes/epel.rb]: We shouldn't need this right? 

EPEL should only be used on Redhat derivatives

File: test/integration/distro/controls/distro_spec.rb [hunk 8]
```diff
@@ -0,0 +1 @@
+include_controls 'install'
```
Comment (damacus) [inline test/integration/distro/controls/distro_spec.rb]: 🎉

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,6 +2,21 @@
 
 This file is used to list changes made in each version of the nginx cookbook.
 
+## 10.0.0-dev (2019-04-09)
```
Comment (JohnRoesler) [inline CHANGELOG.md]: Given this isn't released yet, I'd propose these live under the `Unreleased` heading. And then moved to a `10.0.0` when released

**Comment by fernandoaleman** [comment]
Closing this pull request to open a branch on the `sous-chef` repo. Will send a new PR from there. Will also take into account the comments/suggestions that have been made above.

**Pull request closed without merging**